src/Adapters/ADXAdapter.ts

```diff
@@ -25,6 +33,104 @@ export default class ADXAdapter
         this.adxAuthService.login();
     }
 
+    async getTimeSeriesData(query: string, connection?: IADXConnection) {
+        const adapterMethodSandbox = new AdapterMethodSandbox(
+            this.adxAuthService
+        );
+
+        return await adapterMethodSandbox.safelyFetchData(async (token) => {
+            const clusterUrl = connection
+                ? connection.kustoClusterUrl
+                : this.adxConnectionInformation.kustoClusterUrl;
+
+            if (!isValidADXClusterUrl(clusterUrl)) {
+                adapterMethodSandbox.pushError({
+                    type: ComponentErrorType.DataFetchFailed,
+                    isCatastrophic: true
+                });
+                return new ADXTimeSeriesData(null);
+            }
+
+            const getDataHistoryFromADX = () => {
+                return axios({
+                    method: 'post',
+                    url: `${clusterUrl}/v2/rest/query`,
+                    headers: {
+                        Authorization: 'Bearer ' + token,
+                        Accept: 'application/json',
+                        'Content-Type': 'application/json'
+                    },
+                    data: {
+                        db: connection
+                            ? connection.kustoDatabaseName
```

Comment:
I need to pass connection as optional param here since now every data history widget uses their own connection string and call this method via hook. If not provided, I use the adapters own variables. Think of a scenario like, you created a data history widget using an adt instance X, and by default we use that instance's adx information to set the connection string for that widget. But then, you go an change the selected adt instance in the app, but since scene and widgets are same, I need to rely on the connectionString passed to adapter method here instead of using the adapter's own variables which is tied to the selected adt instance.

Reply:
right. I'm thinking we should just always use the value passed in and never rely on the global scope variable. In what scenario do we not have the value passed in by a widget?